docs/blocks/block-migrations.mdx

```diff
@@ -0,0 +1,206 @@
+This method receives the "old" block (`From` interface) and transforms the block into the new structure (`To` interface).
+
+```ts
+export class ChangeEyebrowMigration extends BlockMigration<(from: From) => To> implements BlockMigrationInterface {
+    ...
+
+    protected migrate({ eyebrow, ...other }: From): To {
+        return {
+            ...other,
+            eyebrow: {
+                draftContent: {
+                    blocks: [
+                        {
+                            key: "12345",
+                            // The "old" eyebrow string
+                            // highlight-next-line
+                            text: eyebrow,
+                            type: "unstyled",
+                            depth: 0,
+                            inlineStyleRanges: [],
+                            entityRanges: [],
+                            data: {},
+                        },
+                    ],
+                    entityMap: {},
+                },
+            },
+        };
+    }
+}
+```
+
+Finally, we add the migration to the block using the `migrate` option in `createBlock`.
+
+```ts
+import { createBlock, typesafeMigrationPipe } from "@comet/blocks-api";
+
+...
+
+export const HeadlineBlock = createBlock(HeadlineBlockData, HeadlineBlockInput, {
+    name: "Headline",
+    migrate: {
+        version: 1,
+        // highlight-next-line
+        migrations: typesafeMigrationPipe([ChangeEyebrowMigration]),
+    },
+});
+```
+
+The next time the block is loaded from the database, the migration will be performed, migrating the block to the new structure.
+
+### Complete migration class
+
+```ts title="1-change-eyebrow.migration.ts"
+import { BlockDataInterface, BlockMigration, BlockMigrationInterface } from "@comet/blocks-api";
+
+interface From {
+    headline: any;
+    eyebrow: string;
+    level: any;
+}
+
+interface To {
+    headline: any;
+    eyebrow: BlockDataInterface;
+    level: any;
+}
+
+export class ChangeEyebrowMigration extends BlockMigration<(from: From) => To> implements BlockMigrationInterface {
+    public readonly toVersion = 1;
+
+    protected migrate({ eyebrow, ...other }: From): To {
+        return {
+            ...other,
+            eyebrow: {
+                draftContent: {
+                    blocks: [
+                        {
+                            key: "12345",
+                            text: eyebrow,
+                            type: "unstyled",
+                            depth: 0,
+                            inlineStyleRanges: [],
+                            entityRanges: [],
+                            data: {},
+                        },
+                    ],
+                    entityMap: {},
+                },
+            },
+        };
+    }
+}
+```
+
+## FAQ
+
+### When do I need a migration?
+
+You will need a migration when
+
+-   **adding** a new field to the block
```

Comment:
is this also true, if the field is optional?

Reply:
No it's not. Haven't thought about that, thanks for the input! Changed in 25fe30c6a769a4389f9d3e835b08bc644fe78b56